docs/source/tutorial/package_testing.rst

```diff
@@ -0,0 +1,198 @@
+extract config that came with the package template, along with the
+``family_and_phenotype.tsv`` data file and extract configuration file
+that you added in the previous step of this tutorial.
+
+Your package probably looks like this so far::
+
+    my_study/
+    ├── data
+    │   ├── clinical.tsv
+    │   └── family_and_phenotype.tsv
+    ├── dataset_ingest_config.yml
+    ├── extract_configs
+    │   ├── extract_config.py
+    │   └── family_and_phenotype.py
+    ├── tests
+    │   ├── conftest.py
+    │   └── test_custom_counts.py
+    └── transform_module.py
+
+
+Try Testing Your Package
+========================
+Let's see what happens if you run:
+
+.. code-block:: bash
+
+    $ kidsfirst test my_study
+
+At the end of the log you should see something like this::
+
+    2019-04-16 10:08:00,101 - kf_lib_data_ingest.app.cli - ERROR - ❌  Ingest pipeline failed validation! See /path/to/my_study/logs/ingest.log for details
+
+And if you scroll up through the log a bit, you should also see::
+
+    EXPECTED COUNT CHECKS
+    +------------------------+------------+---------+---------+
+    | key                    |   expected |   found | equal   |
+    |------------------------+------------+---------+---------|
+    | CONCEPT|FAMILY|ID      |          2 |       2 | ✅      |
+    | CONCEPT|PARTICIPANT|ID |          5 |       9 | ❌      |
+    | CONCEPT|BIOSPECIMEN|ID |         10 |      16 | ❌      |
+    +------------------------+------------+---------+---------+
+
+It looks like the tests failed. Let's break down what is going on and then
+we can update our tests so that they pass.
+
+The ``test`` CLI Command
+========================
+Under the hood, the ``test`` command is really just an alias for:
+
+.. code-block:: shell
+
+    $ kidsfirst ingest <your package> --dry_run
+
+
+The ``test`` command will run ingest on your package without actually loading
+anything into the target service. This will help you quickly determine
+whether your ingest package runs without any exceptions.
+
+Data Validation Tests
+=====================
+Additionally, whenever you run the ``ingest`` command (regardless of whether
+you ran with ``--dry_run`` or not) the data validation tests
+will always execute, allowing you to quickly determine whether or not the
+output of each stage matches a set of accounting expectations you specify in
+your ingest package.
+
+If any one of the test fails, the ingest app will wait until it executes
+all of the tests and then exit with a non-zero exit code.
+The log will indicate what caused the test failure(s).
+
+Standard Counting Tests
+-----------------------------------
+Each stage has a post-run analysis method which performs some basic
+accounting on the stage output and then validates the count results against
+a set of expected counts.
+
+Count Analysis
+^^^^^^^^^^^^^^
+The types of things that are counted are:
+
+- Concepts (i.e. families, participants, biospecimens, etc)
```

Comment:
I _think_ these i.e. should be e.g. but I'm torn because of the 'etc'

Reply:
Lol, i don't know if i've ever checked on what the difference is so I've probably been using i.e. and e.g. completely wrong... I will read and fix 👍